dev/set_matrix.py

```diff
@@ -368,6 +371,75 @@ def get_flavor(name):
     return {"pytorch-lightning": "pytorch"}.get(name, name)
 
 
+def validate_test_coverage(flavor: str, config: Dict):
+    """
+    Validate that all test files for the flavor are executed in the cross-version tests.
+
+    This is done by parsing `run` commands in the `ml-package-versions.yml` to get the list
+    of executed test files, and then comparing it with the actual test files in the directory.
+    """
+    test_dir = os.path.join("tests", flavor)
+    all_test_files = _get_test_files_in_dir(test_dir)
+
+    for category in VALID_CATEGORIES:
+        run_command = config.get(category, {}).get("run")
+        if not run_command:
+            continue
+
+        # Consolidate multi-line commands with "\" to a single line
+        commands = []
+        curr = ""
+        for cmd in run_command.split("\n"):
+            if cmd.endswith("\\"):
+                curr += cmd.rstrip("\\")
+            else:
+                commands.append(curr + cmd)
+                curr = ""
+
+        # Parse pytest commands to get the executed test files
+        for cmd in commands:
+            cmd = cmd.strip().rstrip(";")
+            if cmd.startswith("pytest"):
+                all_test_files -= _get_test_files_from_pytest_command(cmd, test_dir)
+
+    if all_test_files:
```

Comment:
This line looks a bit confusing to me. Instead, can we collect covered files first, and then ___

```python
covered_files = set()

for ...:
    tested_files += ...

if uncovered_test_files := _get_test_files_in_dir(test_dir) - tested_files:
    ...
```